### PR TITLE
Add EtlFileType enum and choice field

### PR DESCRIPTION
**Pull request by HeyZoos**
Adds an enumeration of source file types of datasets represented with the `EtlDataMixin`.

### CODE REVIEW

File: tests/app/migrations/0001_initial.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Migration(migrations.Migration):
                 ('ds_timezone', models.TextField(blank=True, default='UTC', help_text='The timezone applied to date fields.')),
                 ('ds_timerange', django.contrib.postgres.fields.ranges.DateTimeRangeField(blank=True, default=None, help_text='A time range the encompasses the data set date fields.', null=True)),
                 ('ds_configured_date_fields', django.contrib.postgres.fields.jsonb.JSONField(blank=True, default=[], help_text='User configured date fields composed of source fields.')),
-                ('name', models.TextField(help_text='The name of the data set.', unique=True, validators=[plenario_core.validators.starts_with_letter])),
```
Comment (vforgione) [inline tests/app/migrations/0001_initial.py]: Why did all these validators get dropped out?

File: tests/settings.py [hunk 2]
```diff
@@ -68,7 +68,7 @@
         'ENGINE': 'django.contrib.gis.db.backends.postgis',
         'NAME': 'plenario',
         'USER': os.getenv('DJANGO_DB_USER', 'postgres'),
-        'PASSWORD': os.getenv('DJANGO_DB_PASSWORD', ''),
+        'PASSWORD': os.getenv('DJANGO_DB_PASSWORD', 'password'),
```
Comment (vforgione) [inline tests/settings.py]: Why?

File: tests/app/migrations/0001_initial.py (hunk 1 above)
Comment (HeyZoos) [reply tests/app/migrations/0001_initial.py]: Not sure, the migration was generated with the `makemigrations` tool. Is there someway to include that as an option or do I have to edit the migration file myself?

File: tests/settings.py (hunk 2 above)
Comment (HeyZoos) [reply tests/settings.py]: A more convenient default for me that I left in accidentally. I'll remove it.

**Pull request merged**